doc/development/plugins.rst

```diff
@@ -232,6 +232,91 @@ or can include in-built transforms such as:
 * :func:`pennylane.devices.preprocess.no_sampling`
+  
+**Decompositions:**
+
+The :func:`~pennylane.devices.preprocess.decompose` transform is typically required as part of the transform program that
+decomposes unsupported operations to the device's native gate set. To define this transform a stopping condition needs
+to be specified. This is a function mapping an operator to a boolean that determines whether the operator should be decomposed.
+
+For example, for a device supporting the ``CNOT``, ``RX`` and ``RZ`` gates, the stopping condition and the decompose transform
+can be specified like so:
+
+.. code-block:: python
+
+    from pennylane.devices.preprocess import decompose
+
+    def stopping_condition(op):
+        return obj.name in {"CNOT", "RX", "RZ"}
+    
+    program.add_transform(decompose, stopping_condition=stopping_condition, name="my_device")
+
+However, if the device native gate set is unreachable with the default decompositions defined in PennyLane,
+an error will be raised. In this case, you may need to override the decompositions of certain operators
+via the ``decomposer`` argument. 
+
+For example, consider we have a device with ``RX``, ``RY`` and ``IsingXX`` as native gates but we want
+to execute a circuit written in terms of ``CNOT`` s. Then, we can define a decomposition for ``CNOT`` 
+(e.g., ``custom_decomposer``) and pass it to the decomposer kwarg:
+
+.. code-block:: python
+
+    def stopping_condition(op):
+        return op.name in {"IsingXX", "RX", "RY"}
+
+    def custom_decomposer(op):
+        if isinstance(op, qml.CNOT):
+            wires = op.wires
+            return [
+                qml.RY(np.pi/2, wires=wires[0]),
+                qml.IsingXX(np.pi/2, wires=wires),
+                qml.RX(-np.pi/2, wires=wires[0]),
+                qml.RY(-np.pi/2, wires=wires[0]),
+                qml.RY(-np.pi/2, wires=wires[1])
+            ]
+        return op.decomposition()
+    
+    program.add_transform(
+        decompose,
+        stopping_condition=stopping_condition,
+        decomposer=custom_decomposer,
+        name="my_device"
+    )
+
+An alternative approach for overriding the decompositions of certain operators is by taking advantage of
+the new graph-based decomposition algorithm, with :func:`qml.decomposition.enable_graph() <~pennylane.decomposition.enable_graph>`.
+In this case the ``target_gates`` kwarg needs to be specified in the :func:`~pennylane.devices.preprocess.decompose`
+transform. Note that the stopping condition function defines whether an operator should be decomposed,
+while the ``target_gates`` defines the set of operator types that the graph-based decomposition
+algorithm needs to target.
+
+With this approach, the decomposition for the CNOT needs to be specified as a quantum function, ``decompose_cnot``, and
+registered with ``qml.add_decomps``:
+
+.. code-block:: python
+
+    qml.decomposition.enable_graph()
```

Comment:
This can be slightly misleading. The device developer wouldn't be enabling the graph within its preprocess program. The device developer needs to set up the preprocess program in a way that makes it work when graph is enabled by the user.